syntax = "proto3";

package workerpackage;

service Worker {
    rpc SetDriverPort(DriverPort) returns (Status);
    rpc Map(MapInput) returns (Status);
    rpc Reduce(ReduceId) returns (Status);
    rpc Terminate(Empty) returns (Status);
}

message DriverPort {
    int32 port = 1;
}

message Status {
    string msg = 1;
    int32 code = 2;
}

message MapInput {
    string file_path = 1;
    int32 map_task_id = 2;
    int32 m = 3;
}

message ReduceId {
    int32 reduce_task_id = 1;
}

message Empty {}


